updater/lib/dependabot/updater/group_update_creation.rb

```diff
@@ -232,6 +238,32 @@ def peer_dependency_should_update_instead?(dependency_name, dependency_files, up
               can_update?(requirements_to_unlock: :own)
           end
       end
+
+      def prepare_workspace
+        return unless job.clone? && job.repo_contents_path
+
+        # TODO: Remove the directory parameter
+        #
+        # We should defer calculation of the `path` to the call site in the shared_helper
+        # so it is impossible to calculate different values for workspace and non-workspace
+        # calls to the helper.
```

Comment:
Good callout 👍

Reply:
This might be a bit of a 🐔 and 🥚 having looked at it more as we rely on the path to set the git configuration within the repo folder - setting them globally seems like bad form and I'd prefer not to - but it's mostly a side effect of us passing the `repo_contents_path` and `directory` right down the stack to the native helper.

They should always be the same in both call-sites as they do not change for the entire context of a run[^1], so I think while it feels odd as they could in principal be out of sync that shouldn't ever happen without a bug being introduced.

[^1]: Unless we support cross-folder grouping jobs in future 🤔

Reply:
I didn't say the important part in my last comment - I'm going to back off changing this for now